### PR TITLE
Deprecate `box` in favor of `div`

### LINKED ISSUE #379: Consider to deprecate box component
As standard HTML tags can now (Issue #337) be styled using the styling DSL, we could consider to deprecate the ``box`` component, which is just a styled div!
```kotlin
// before 0.10
box({
    // some styling DSL
}) {...}

// starting with 0.10. absolutly equivalent
div({
    // some styling DSL
}) {...}
```

### PR DESCRIPTION
**Pull request by haukesomm**
This PR deprecates the `box` factory method in favor of `div` because, in combination with fritz2.styling, it offers the exact same functionality. All calls of `box` will have to be replaced eventually.

Closes #379 .

### CODE REVIEW

File: components/src/jsMain/kotlin/dev/fritz2/components/box.kt [hunk 1]
```diff
@@ -33,6 +33,13 @@ import dev.fritz2.styling.params.GridParams
  * @param prefix the prefix for the generated CSS class resulting in the form ``$prefix-$hash``
  * @param content a lambda expression for setting up the content and events of the ``div`` element itself
  */
+@Deprecated(
```
Comment (chausknecht) [inline components/src/jsMain/kotlin/dev/fritz2/components/box.kt:36]: I think the replacement expression is far too complicated and wrong:
```kotlin
box({
    background { color { "purple" } }
}) {
    +"Test"
}
```
If applied the above becomes to:
```kotlin
div(fun FlexParams.() {
    background { color { "purple" } }
}, StyleClass.None, null, "box") {
    +"Test"
}
```
Definitly not correct!

Should it not simplex be a chnage from `box` to `div` like this?
```kotlin
div({
    background { color { "purple" } }
}) {
    +"Test"
}
```
The import is nice of course!

**Review by chausknecht — CHANGES_REQUESTED** [review]
Correct replacement expression if my assumptions is correct!

**Comment by chausknecht** [comment]
I think a corresponding PR for the KitchenSink is yet missing - lots of boxes left there ;-)

File: components/src/jsMain/kotlin/dev/fritz2/components/box.kt (hunk 1 above)
Comment (haukesomm) [reply components/src/jsMain/kotlin/dev/fritz2/components/box.kt:36]: Afaik `ReplaceWith` does not have full support for both optional parameters and lambdas which results in the generated replacement being overly complicated, yet technically correct. The only other option that comes to my mind is to get rid of the `ReplaceWith` as the conversion is simple enough to just be mentioned in the release notes 🤔

**Review by chausknecht — APPROVED** [review]

**Pull request merged**